src/lib/hooks/swap/useSwapInfo.tsx

```diff
@@ -52,6 +53,10 @@ function useComputeSwapInfo(): SwapInfo {
     useMemo(() => [inputCurrency ?? undefined, outputCurrency ?? undefined], [inputCurrency, outputCurrency])
   )
 
+  // Short circuit trade fetching and amount parsing if wrapping or unwrapping.
+  const { type: wrapType } = useWrapCallback()
+  const isWrapping = wrapType === WrapType.WRAP || wrapType === WrapType.UNWRAP
```

Comment:
@zzmp do i need to memoize this?

Reply:
no, it's a primitive type (boolean), so not memoizing will not cause any extra re-renders